src/storage/user-data.ts

```diff
@@ -67,13 +140,16 @@ export async function loadUserData(): Promise< UserData > {
 		const asString = await fs.promises.readFile( filePath, 'utf-8' );
 		try {
 			const parsed = JSON.parse( asString );
-			const data = fromDiskFormat( parsed );
+			let data = fromDiskFormat( parsed );
 
 			// Temporarily populate old snapshots with userId of authenticated user.
 			// See PR #937 for more context.
 			legacyPopulateSnapshotUserIds( data );
 			sortSites( data.sites );
 			populatePhpVersion( data.sites );
+
+			data = await migrateWpAutoUpdating( data );
```

Comment:
I think this function should be executed only once and not every time Studio starts.

Reply:
I was thinking about it and why I went this approach:
1. to check - should we run this function, we should do at least 'deta.sites.filter()', so it means that we should run at least 1 loop always.
2. With the current approach, if all sites are immigrated, we run only one loop, so totally the same https://github.com/Automattic/studio/pull/1189/files#diff-2a78c3d26284d759ad9ec39370822d96d2dcbf38c383487f49c15f67192682daR85

So it's trade of - extra `filter` function outside or `map` to create new array inside. performance will be totally the same.

So it's only about readability and thinking about it... probably you are right, condition outside would keep better readability 👍 I will add it

Reply:
Done